### PR TITLE
[Onboarding]: NV-3415 | Feature flag and conditional rendering setup

### PR DESCRIPTION
**Pull request by antonjoel82**
### What change does this PR introduce?

- Refactors the top-level `App` to separate it from the `Routes` and non-React initialization steps
- Adds the feature flag for Onboarding and related hook setup
- Little high-level tweaks for `Get started` page
- Stylistic tweaks to the timeline + semantic HTML updates

### Why was this change needed?

- Refactoring was necessary to enable us to use hooks and feature flags to conditionally handle routes
  - Furthermore, this provides clearer boundaries for App organization in line with the Single Responsibility principle
- Solves [NV-3415](https://linear.app/novu/issue/NV-3415/🙋-onboarding-feature-flag)

### Other information

- There is now proper coloring of the bullets and timeline connector, along with line heights of the text paragraphs
Screenshot 2024-02-08 at 1 09 02 PM- _The new onboarding flow will appear all the time for now because the default value of the hook is true_ 

### Validation steps

1. Check-out the branch and login
2. Navigate to the `Get started` page
3. The new (non-functional) flow should still appear

### CODE REVIEW

File: apps/web/src/App.tsx [hunk 1]
```diff
@@ -1,255 +1,10 @@
...
-     * Set tracesSampleRate to 1.0 to capture 100%
-     * of transactions for performance monitoring.
-     * We recommend adjusting this value in production
-     */
-    tracesSampleRate: 1.0,
-    beforeSend(event: Sentry.Event) {
-      return event;
-    },
-  });
-}
```
Comment (antonjoel82) [inline apps/web/src/App.tsx:104]: Moved to `initializeApp.ts`

File: apps/web/src/App.tsx [hunk 2]
```diff
@@ -1,255 +1,10 @@
...
-applyToken(tokenStoredToken);
-
-function App() {
+export default function App() {
   return (
-    <SegmentProvider>
-      <HelmetProvider>
-        <BrowserRouter basename={CONTEXT_PATH}>
-          <QueryClientProvider client={queryClient}>
-            <AuthProvider>
```
Comment (antonjoel82) [inline apps/web/src/App.tsx:130]: Moved to `Providers.tsx`

File: apps/web/src/App.tsx [hunk 3]
```diff
@@ -1,255 +1,10 @@
...
-                  <Route path={ROUTES.TEAM} element={<MembersInvitePage />} />
-                  <Route path={ROUTES.CHANGES} element={<PromoteChangesPage />} />
-                  <Route path={ROUTES.SUBSCRIBERS} element={<SubscribersList />} />
-                  <Route path={ROUTES.BRAND} element={<BrandPage />}>
-                    <Route path="" element={<BrandingForm />} />
-                    <Route path="layouts" element={<LayoutsListPage />} />
-                  </Route>
-                  <Route path="/translations/*" element={<TranslationRoutes />} />
-                </Route>
-              </Routes>
```
Comment (antonjoel82) [inline apps/web/src/App.tsx:239]: Moved to `AppRoutes.tsx`

Separating theses from the Providers enables us to use feature flags at the route-level, which is great for large page change features like this

File: apps/web/src/pages/get-started/consts/shared.tsx [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 import { colors } from '@novu/design-system';
 
-export const StepText = styled.div`
+export const StepText = styled.p`
   display: inline;
   color: ${colors.B60};
 `;
 
-export const StepDescription = styled.p`
+export const StepDescription = styled.div`
```
Comment (antonjoel82) [inline apps/web/src/pages/get-started/consts/shared.tsx:9]: Was getting a warning about illegal DOM node nesting

File: apps/web/src/App.tsx (hunk 3 above)
Comment (LetItRock) [reply apps/web/src/App.tsx:239]: finally! thank you so much Joel! 🔥

**Review by LetItRock — APPROVED** [review]
love these changes! 🙌

**Pull request merged**